crates/polars-plan/src/dsl/mod.rs

```diff
@@ -435,6 +436,23 @@ impl Expr {
         self.apply_many_private(FunctionExpr::TopK(false), &[k], false, false)
     }
 
+    /// Returns the `k` largest rows by given column.
+    ///
+    /// For single column, use [`Expr::top_k`].
+    #[cfg(feature = "top_k")]
+    pub fn top_k_by<K: Into<Expr>, E: AsRef<[IE]>, IE: Into<Expr> + Clone, G: AsRef<[bool]>>(
+        self,
+        k: K,
+        by: E,
+        descending: G,
+    ) -> Self {
+        self.sort_by(
```

Comment:
I've noticed this. Currently it is just implemented the same as `LazyFrame.top_k`. Can we just merge this and I will open another PR to improve both?

Reply:
It should be in this PR. The `LazyFrame.top_k` will be optimized to a specialized `top-k`, this will not.  A `top-k` should not do a full sort. If we support something, I want us to do it properly.